lib/widgets/like_counter.dart

```diff
@@ -0,0 +1,67 @@
+import 'package:flutter/material.dart';
+import '../../../utils/ui_constants.dart';
+
+class LikeCounter extends StatefulWidget {
+  static const double iconSize = UIConstants.posterSectionIconSize;
+  static const String buttonText = 'Likes: ';
+
+  const LikeCounter({super.key});
+
+  @override
+  State<StatefulWidget> createState() => _LikeCounterState();
+}
+
+class _LikeCounterState extends State<LikeCounter> {
+  static const buttonTextStyle = TextStyle(
+    color: Colors.white,
+    fontWeight: FontWeight.bold,
+    fontSize: 20,
+  );
+  int _likeCount = 0;
+  _LikeCounterState();
+
+  void _incrementLikes() {
+    setState(() {
+      _likeCount++;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Stack(
+      alignment: Alignment.topCenter,
+      children: [
+        Container(
+          height: LikeCounter.iconSize,
+          width: LikeCounter.iconSize,
+          decoration: const BoxDecoration(
+            color: Colors.redAccent,
+            shape: BoxShape.circle,
+          ),
+          child: Align(
+            alignment: Alignment.center,
+            child: Column(
+              mainAxisAlignment: MainAxisAlignment.center,
+              children: [
+                const Text(
+                  textAlign: TextAlign.center,
+                  LikeCounter.buttonText,
+                  style: buttonTextStyle,
+                ),
+                Text(
+                  textAlign: TextAlign.center,
+                  '$_likeCount',
+                  style: buttonTextStyle,
+                ),
+              ],
+            ),
+          ),
+        ),
+        ElevatedButton(
+          onPressed: _incrementLikes,
+          child: const Icon(Icons.add_circle),
+        ),
```

Comment:
Do you really need this button to increase likes count? Maybe you can use the Container and wrapped into a Inkwell

Reply:
It does look a lot better, thank you!